Outdated Github Actions CI pnpm version

### Current Behavior

After creating a nx workspace using pnpm (v9.9.0) and github actions the CI fails due to: `WARN  Ignoring not compatible lockfile`. The reason why it fails is because the pnpm version in the workflow is set to 8 and does not support the pnpm v9 lockfile.

### Expected Behavior

The CI workflow should support pnpm latest version lockfile.

### GitHub Repo

https://github.com/juliangsibecas/nx-ci-poc

### Steps to Reproduce

1. npx create-nx-workspace@latest my-workspace --pm pnpm
2. git remote add origin git@github.com:juliangsibecas/nx-ci-poc.git
3. git push origin main


### Nx Report

```shell
Node           : 18.18.2
OS             : darwin-arm64
Native Target  : aarch64-macos
pnpm           : 9.9.0

nx (global)    : 19.6.4
nx             : 19.6.5
@nx/js         : 19.6.5
@nx/workspace  : 19.6.5
```


### Failure Logs

```shell
pnpm install --frozen-lockfile
  shell: /usr/bin/bash -e {0}
  env:
    PNPM_HOME: /home/runner/setup-pnpm/node_modules/.bin
 WARN  Ignoring not compatible lockfile at /home/runner/work/nx-ci-poc/nx-ci-poc/pnpm-lock.yaml
 ERR_PNPM_NO_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is absent
```


### Package Manager Version

_No response_

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/packages/workspace/src/generators/ci-workflow/files/github/.github/workflows/__workflowFileName__.yml__tmpl__
+++ b/packages/workspace/src/generators/ci-workflow/files/github/.github/workflows/__workflowFileName__.yml__tmpl__
@@ -20,7 +20,7 @@ jobs:
       <% if(packageManager == 'pnpm'){ %>
       - uses: pnpm/action-setup@v4
         with:
-          version: 8
+          version: 9
       <% } %>
       <% if(packageManager == 'bun'){ %>
       - uses: oven-sh/setup-bun@v1

